daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto

```diff
@@ -1184,49 +1091,23 @@ message CaseAlt {
     // name of the type constructor
     TypeConName con = 1;
 
-    // name of the variant constructor
-    // *Must be a valid identifier*
-    oneof constructor {
-      // *Must be a valid identifier*
-      string constructor_str = 2; // *Available in version < 1.7*
-
-      // *Must be a valid interned identifier*
-      int32 constructor_interned_str = 3; // *Available in versions >= 1.7*
-    }
+    // *Must be a valid interned identifier*
+    int32 constructor_interned_str = 3;
   }
 
   // Non empty list pattern
   message Cons {
+    // *Must be a valid interned identifier*
+    int32 var_head_interned_str = 3;
 
-    // name of the binder for the head
-    oneof var_head {
-      // *Must be a valid identifier*
-      string var_head_str = 1; // *Available in version < 1.7*
-
-      // *Must be a valid interned identifier*
-      int32 var_head_interned_str = 3; // *Available in versions >= 1.7*
-    }
-
-
-    oneof var_tail {
-      // *Must be a valid identifier*
-      string var_tail_str = 2; // *Available in version < 1.7*
-
-      // *Must be a valid interned identifier*
-      int32 var_tail_interned_str = 4; // *Available in versions >= 1.7*
-    }
+    // *Must be a valid interned identifier*
+    int32 var_tail_interned_str = 4;
   }
 
   // Non empty option patterm
   message OptionalSome {
-
-    oneof var_body {
-      // *Must be a valid identifier*
-      string var_body_str = 1; // *Available in version < 1.7*
-
       // *Must be a valid interned identifier*
-      int32 var_body_interned_str = 2; // *Available in versions >= 1.7*
-    }
+      int32 var_body_interned_str = 2;
```

Comment:
can we renumber the fields too? or do you want to do it later?

```suggestion
      int32 var_body_interned_str = 1;
```

Reply:
We can. 
I just plan to do all the filed renumbers at the end.